File: app/src/main/java/com/worldventures/dreamtrips/modules/friends/service/command/GetUsersCommand.java
package com.worldventures.dreamtrips.modules.friends.service.command;


import com.worldventures.dreamtrips.core.api.action.CommandWithError;
import com.worldventures.dreamtrips.core.janet.JanetModule;
import com.worldventures.dreamtrips.core.janet.dagger.InjectableAction;
import com.worldventures.dreamtrips.modules.common.model.User;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Named;

import io.techery.janet.Janet;
import io.techery.mappery.MapperyContext;

public abstract class GetUsersCommand extends CommandWithError<List<User>> implements InjectableAction {

   @Inject @Named(JanetModule.JANET_API_LIB) Janet janet;
   @Inject MapperyContext mapperyContext;

   protected List<User> convert(Iterable<?> itemsToConvert) {
      return mapperyContext.convert(itemsToConvert, User.class);
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/users/base/command/GetUsersCommand.kt
package com.worldventures.dreamtrips.social.service.users.base.command

import com.worldventures.core.janet.CommandWithError
import com.worldventures.core.model.User
import com.worldventures.janet.injection.InjectableAction
import io.techery.janet.Janet
import io.techery.mappery.MapperyContext
import javax.inject.Inject

abstract class GetUsersCommand : CommandWithError<List<User>>(), InjectableAction {

   @Inject lateinit var janet: Janet
   @Inject lateinit var mapperyContext: MapperyContext

   protected open fun convert(itemsToConvert: Iterable<*>) = mapperyContext.convert(itemsToConvert, User::class.java)

}
